src/java.desktop/share/classes/sun/awt/AWTAccessor.java

```diff
@@ -863,6 +877,23 @@ void setDropTargetContextPeer(DropTargetContext dtc,
     private static DragSourceContextAccessor dragSourceContextAccessor;
     private static DropTargetContextAccessor dropTargetContextAccessor;
 
+    /*
+     * Set an accessor object for the java.awt.color.ICC_Profile class.
+     */
+    public static void setICC_ProfileAccessor(ICC_ProfileAccessor ipa) {
+        iccProfileAccessor = ipa;
+    }
+
+    /*
+     * Retrieve the accessor object for the java.awt.color.ICC_Profile class.
+     */
+    public static ICC_ProfileAccessor getICC_ProfileAccessor() {
+        if (iccProfileAccessor == null) {
```

Comment:
For `SharedSecrets` in java.base code was updated to use single read (to avoid concurrency problems) - see [JDK-8259021](https://bugs.openjdk.org/browse/JDK-8259021)
Shouldn't we use the same patter here?

Reply:
Yes, you are right, let me fix that first.

Reply:
The AWTAccessor and SwingAccessor files are updated by https://github.com/mrserb/jdk/commit/c50a9047b4324b280b2b5d1fc70e18aac372a50c. This patch now uses the new pattern.